dashboard/app/helpers/users_helper.rb

```diff
@@ -2,6 +2,12 @@ module UsersHelper
 
   # Summarize the current user's progress within a certain script.
   def summarize_user_progress(script, user = current_user)
+    # Cache the results in an instance variable (per-request)
+    # todo: investigate using Rails.cache with appropriate user+script cache keys instead
+    @user_data ||= {}
```

Comment:
hm, I'd be a bit wary of using an instance variable here since UsersHelper is a module so it's hard to predict what class that instance variable is going to end up in.

Is calling summarize_user_progress multiple times actually a problem for the ops_data_pull script? as far as I can tell you're already storing the stuff you care about in the level_progress local variable so it never gets called twice with the same script and user?